File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
def dfs(matrix:list[list[int]],visited : list[bool],vertex : int):
    if(visited[vertex] == True):
        return
    # 방문 하지 않았다
    visited[vertex] = True # 방문했다고 표기
    # 방문 했던 점이 아니면 경로에 추가
    print(vertex+1, end=" ")
    for index in range(len(matrix)):
        if matrix[vertex][index] == 1:
            dfs(matrix,visited,index)

def bfs(matrix:list[list[int]],visited : list[bool],vertex : int):
    # 큐 자료 구조 활용
    queue = []
    queue.append(vertex)
    visited[vertex] = True

    while(len(queue) != 0):
        ## 큐에 먼저 추가된 요소 뽑고 출력
        cur = queue.pop(0)
        print(cur+1,end=" ")
        ## 현재 레벨에서 방문하지 않았고 갈수 있는 모든 점
        for i in range(len(matrix)):
            if(matrix[cur][i] == 1 and (not visited[i])):
                queue.append(i)
                visited[i] = True
    return 

N,M,V=map(int, input().split())
Matrix = [[0 for j in range(N)]for i in range(N)]
for i in range(M):
    From, To = map(int,input().split())
    Matrix[From-1][To-1] = 1
    Matrix[To-1][From-1] = 1

visited = [False for i in range(N)]
dfs(Matrix,visited,V-1)
print()
visited = [False for i in range(N)]
bfs(Matrix,visited,V-1)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java

import java.io.*;
import java.util.*;
public class Main{
    static List<List<Integer>> graph = new ArrayList<>();
    static boolean[] visited;
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        //BufferedReader br = new BufferedReader(new FileReader("input.txt"));
        // 카드의 색 : R,B,Y,G
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int N,M,V;
        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());
       
        for(int i=0; i<=N; i++){
            graph.add(new ArrayList<>());
        }
        for(int i=0; i<M; i++){
            st = new StringTokenizer(br.readLine());
            int start = Integer.parseInt(st.nextToken());
            int end = Integer.parseInt(st.nextToken());
            // 무방향 그래프
            graph.get(start).add(end);
            graph.get(end).add(start);
        }
        for (List<Integer> g : graph){
            g.sort(Comparator.naturalOrder());
        }
        visited = new boolean[N+1];
        dfs(V);
        visited = new boolean[N+1];
        System.out.println();
        bfs(V);
    }
    private static void bfs(int v){
        Deque<Integer> queue = new ArrayDeque<>();
        queue.addLast(v);
        visited[v] = true;
        while(!queue.isEmpty()){
            int current = queue.removeFirst();
            System.out.print(current + " ");
            for (Integer next : graph.get(current)) {
                if (!visited[next]){
                    queue.addLast(next);
                    visited[next] = true;
                }
            }
        }
    }
    private static void dfs(int v){
        if(visited[v] == true) return;
        visited[v] = true;
        System.out.print(v + " ");
        for (Integer next : graph.get(v)){
            if (!visited[next]) dfs(next);
        }
    }
}